src/Workspaces/Core/Portable/Formatting/Engine/TokenStream.cs

```diff
@@ -501,9 +530,35 @@ private bool IsFirstTokenOnLine(TokenData tokenData1, TokenData tokenData2)
             return this.GetTriviaData(tokenData1, tokenData2).SecondTokenIsFirstTokenOnLine;
         }
 
+        /// <summary>
+        /// An approximation for the length of a token in this stream
+        /// </summary>
+        float AverageTokenLength { get; }
```

Comment:
1. can you move this up to the top with the rest of the field/props.
2. can you use double here.

Reply:
Moved. Any reason to use `double`?

Reply:
double is the 'standard' floating point type for .net and c#.  if you write 0.0, you're getting a double by default.  In effect, it's what should normally be used unless there is a good reason to move away from it (same with 'int' versus any of the other integral types).  Thanks, and good question :)

Reply:
I decided to keep `float` for now. In this case even fixed-point number with 3-4 bits for the fractional part would be enough. No need to waste space.